### ISSUE TITLE
Implement Warning/Error filters using the total warnings/errors calculated in the active mod list

### ISSUE THREAD

**Issue by WAGEDATWAR** [RimSort/RimSort#297]
**Is your feature request related to a problem? Please describe.**
There doesn't appear to be a way to only show mods within a list that have warnings or errors, or sort them to the top, for easier viewing. This makes a number of update and pack maintenance tasks much more annoying to perform and has been a feature in Rimpy for longer than I've used it. This is especially necessary as we're waiting for many mods to update to 1.5 and will be for a while.

**Describe the solution you'd like**
The best way to implement this without altering the design language would be making the warning number notification on the bottom bar a button that only show mods that have warnings in the active modlist. This same logic should also apply to the error notification. Visual feedback when the cursor hovers over them would also be a nice touch.

**Describe alternatives you've considered**
Idk really, this is a fairly ubiquitous feature and I was surprised it isn't already implimented.

**Additional context**
(If these are already there it makes sense to have them act as filters.)
![Screenshot_456](https://github.com/RimSort/RimSort/assets/63970392/68db614c-4839-4f08-a6c0-60d9061fca3a)

**Comment by twstagg** [comment]
I agree. This should definitely be a feature. The metadata is there. Just need to make the UI able to use it in this manner.
